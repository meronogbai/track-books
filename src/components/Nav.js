import {
  faChartPie, faEllipsisH, faPlus, faPoll,
} from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const { Link } = require('react-router-dom');

const Nav = () => (
  <nav className="Nav">
    <Link to="/add-book">
      <FontAwesomeIcon icon={faPlus} size="lg" />
      <span>Add Book</span>
    </Link>
    <Link to="/track">
      <FontAwesomeIcon icon={faPoll} size="lg" />
      <span>Track</span>
    </Link>
    <Link to="/home">
      <FontAwesomeIcon icon={faChartPie} size="lg" />
      <span>Progress</span>
    </Link>
    <Link to="/home">
      <FontAwesomeIcon icon={faEllipsisH} size="lg" />
      <span>More</span>
    </Link>
  </nav>
);

export default Nav;
