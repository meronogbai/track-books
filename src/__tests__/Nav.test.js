import { create } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Nav from '../components/Nav';

test('renders correctly', () => {
  const tree = create(<BrowserRouter><Nav /></BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});
