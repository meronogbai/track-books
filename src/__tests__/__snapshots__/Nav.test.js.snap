// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="Nav"
>
  <a
    href="/add-book"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-plus fa-w-14 fa-lg "
      data-icon="plus"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span>
      Add Book
    </span>
  </a>
  <a
    href="/track"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-poll fa-w-14 fa-lg "
      data-icon="poll"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zM160 368c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16V240c0-8.84 7.16-16 16-16h32c8.84 0 16 7.16 16 16v128zm96 0c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16V144c0-8.84 7.16-16 16-16h32c8.84 0 16 7.16 16 16v224zm96 0c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16v-64c0-8.84 7.16-16 16-16h32c8.84 0 16 7.16 16 16v64z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span>
      Track
    </span>
  </a>
  <a
    href="/progress"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-chart-pie fa-w-17 fa-lg "
      data-icon="chart-pie"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 544 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M527.79 288H290.5l158.03 158.03c6.04 6.04 15.98 6.53 22.19.68 38.7-36.46 65.32-85.61 73.13-140.86 1.34-9.46-6.51-17.85-16.06-17.85zm-15.83-64.8C503.72 103.74 408.26 8.28 288.8.04 279.68-.59 272 7.1 272 16.24V240h223.77c9.14 0 16.82-7.68 16.19-16.8zM224 288V50.71c0-9.55-8.39-17.4-17.84-16.06C86.99 51.49-4.1 155.6.14 280.37 4.5 408.51 114.83 513.59 243.03 511.98c50.4-.63 96.97-16.87 135.26-44.03 7.9-5.6 8.42-17.23 1.57-24.08L224 288z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span>
      Progress
    </span>
  </a>
</nav>
`;
